src/util/os_utils.js

```diff
@@ -591,16 +592,14 @@ function is_folder_permissions_set(current_path) {
 }
 
 function read_server_secret() {
-    if (process.env.SERVER_SECRET) {
-        return process.env.SERVER_SECRET;
-    } else {
-        // in kubernetes we must have SERVER_SECRET loaded from a kubernetes secret
-        if (process.env.CONTAINER_PLATFORM === 'KUBERNETES') {
-            throw new Error('SERVER_SECRET env variable not found. it must exist when running in kubernetes');
-        }
-        // for all non kubernetes platforms (docker, local, etc.) return a dummy secret
-        return '12345678';
+    const server_secret = ssl_utils.get_data_from_kubernetes_secret(`/etc/noobaa-server`, `server_secret`);
+    if (server_secret) return server_secret;
+    // in kubernetes we must have SERVER_SECRET loaded from a kubernetes secret
+    if (process.env.CONTAINER_PLATFORM === 'KUBERNETES') {
+        throw new Error('server_secret mounted to a file was not found. it must exist when running in kubernetes');
     }
+    // for all non kubernetes platforms (docker, local, etc.) return a dummy secret
+    return '12345678';
```

Comment:
Why do we send a different dummy secret?
`'12345678'` from `read_server_secret`, while `'abcdefgh'` from `get_jwt_secret()`

Reply:
This is not in the scope of this PR, and I don't want to challenge it here.
This difference was probably created to diverse between server secret and jwt